testsuite/integration-arquillian/servers/auth-server/jboss/pom.xml

```diff
@@ -416,16 +416,19 @@
                                     <configuration>
                                         <target>
                                             <ant antfile="${common.resources}/ant/configure.xml" target="configure-server-jpa">
+                                                <!-- I'm intentionally leaving this as a comment. Before this gets integrated,
+                                                     I need to speak with Tomas and Vlasta if their use cases work -->
+
                                                 <!-- These properties become equivalent to properties defined on the command line. -->
                                                 <!-- Without specifying those the default values would be used regardless what is -->
                                                 <!-- defined via -Dproperty=value when executing maven command -->
-                                                <property name="jdbc.driver.tmp.dir">${jdbc.driver.tmp.dir}</property>
-                                                <property name="jdbc.mvn.artifactId">${jdbc.mvn.artifactId}</property>
-                                                <property name="jdbc.mvn.version">${jdbc.mvn.version}</property>
-                                                <property name="keycloak.connectionsJpa.url">${keycloak.connectionsJpa.url}</property>
-                                                <property name="keycloak.connectionsJpa.user">${keycloak.connectionsJpa.user}</property>
-                                                <property name="keycloak.connectionsJpa.password">${keycloak.connectionsJpa.password}</property>
-                                                <property name="keycloak.connectionsJpa.schema">${keycloak.connectionsJpa.schema}</property>
+<!--                                                <property name="jdbc.driver.tmp.dir" value="${jdbc.driver.tmp.dir}" />-->
```

Comment:
@vramik @tkyjovsk Here goes a very tricky part and I would like to know your opinion about it. However, before diving into my explanation, please read these two articles:

* https://technotes.khitrenovich.com/properties-resolution-maven-implications-antrun-plugin/
* https://books.sonatype.com/mvnref-book/reference/resource-filtering-sect-properties.html

The comment here suggests that you have been struggling with property resolution here. I also encountered some difficulties, because Maven resolves expressions when rendering the Effective POM. Since the DB Allocator Plugin manipulates `keycloak.connectionsJpa.*` properties in the runtime, resolving them for the Ant Plugin happened too quickly. This resulted in picking up properties defined [here](https://github.com/keycloak/keycloak/blob/31b767567e2c2940766c054a3298b73cfa8a7ec5/testsuite/integration-arquillian/pom.xml#L91-L96), instead of those changed by the DB Allocator Plugin.

The solution I propose is to force the Ant Plugin to resolve the properties instead of relying on the mechanism provided by Maven. Here's an example:

Maven resolution example:
```xml
<property name="jdbc.driver.tmp.dir">${jdbc.driver.tmp.dir}</property>
```

Ant Plugin resolution example:
```xml
<property name="jdbc.driver.tmp.dir" value="${jdbc.driver.tmp.dir}" />
```

Note, that the plugin now receives ''${jdbc.driver.tmp.dir}" and needs to resolve this based on the properties passed by Maven. So this time, the resolution happens in the runtime instead of rendering an Effective POM.

I'm not entirely sure if I didn't break anything with this change. So if you guys are invoking something manually somewhere, please check if it still works.

Reply:
If I understand it correctly the mechanisms you're proposing is already used e.g. [here](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/servers/auth-server/jboss/pom.xml#L219-L229)

Reply:
Yes, as you can see, this Pull Request replaces the piece you mentioned in your comment. In other words, instead of using:

```xml
<property name="jdbc.driver.tmp.dir">${jdbc.driver.tmp.dir}</property>
```

We will be using:

```xml
<property name="jdbc.driver.tmp.dir" value="${jdbc.driver.tmp.dir}" />
```

But as I said, I potentially might have broken something (please keep in mind that I'm in the team for quite a short time). So if you're using this properties somewhere (e.g. in prod testing or whatever), could you please run it with this patch to see if everything is still fine?

Reply:
If the overriding of the property from the command line still works: `mvn clean install -Djdbc.driver.tmp.dir=/my/favourite/folder/` it's fine to do the change you're proposing.

Reply:
I've just checked it and it seems to work fine.